### ISSUE TITLE
[prefer-function-type] fix generates invalid code in union or intersection types

### ISSUE THREAD

**Issue by tao-cumplido** [typescript-eslint/typescript-eslint#3000]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-function-type": ["error"]
  }
}
```

```TS
type X = {} | {
  (): void;
}
```

**Expected Result**

```ts
type X = {} | (() => void)
```

**Actual Result**

```ts
type X = {} | (() => void;) // ')' expected. ts(1005)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.2` |
| `@typescript-eslint/parser`        | `4.14.2` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.19.0` |
| `node`                             | `14.15.4` |
